geomstats/geometry/fiber_bundle.py

```diff
@@ -8,7 +9,7 @@
 from geomstats.geometry.riemannian_metric import RiemannianMetric
 
 
-class FiberBundle(Manifold):
+class FiberBundle(Manifold, ABC):
```

Comment:
Also, maybe this class should be renamed ```PrincipalBundle``` at some point ? Or if this is supposed to be more general, allow for other types of fibers such as vector spaces ?

Reply:
For your first question, I hesitated and tried both. 
The first thing to decide is whether we want FiberBundle to represent a space at all. In the previous versions I had it identified with the total space (but it was not very clear). This avoids creating another object: one for the total space, one for the base, one for the bundle etc. The current implementation of the `PreShapeSpace` does this identification. In this case we need `FiberBundle` to inherit from `Manifold`, to make sure it has the attributes (dim, default_point_type) and implements the usual methods a manifold must have (belongs, to_tangent etc)

Reply:
For the second question, I think `FiberBundle` is more general and could exist if we have a submersion that is not necessarily defined by a group action

Reply:
To me, if ```FiberBundle``` were to represent a space, it should be the base manifold (and I see that ```belongs``` checks in priority that a point belongs to the base manifold), because the fiber bundle structure allows to define a geometry on the base space (from (1) the total space, (2) a (group) action and (3) a Riemannian metric on the total space that is invariant wrt the action). I'm guessing this is what you had in mind ? Maybe renaming ```FiberBundle``` into ```QuotientSpace``` would make it more clear ? Then we would have the 3 abstract or base classes that you and @ninamiolane were discussing: ```EmbeddedManifold```, ```OpenSet``` and ```QuotientSpace``` ?

Reply:
Or that would actually make 4 abstract classes with ```Manifold``` ?